homeassistant/components/aladdin_connect/config_flow.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aladdin_connect import AladdinConnectClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    acc = AladdinConnectClient(data[CONF_USERNAME], data[CONF_PASSWORD])
+    try:
+        login = await hass.async_add_executor_job(acc.login)
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        raise ConnectionError from ex
+    else:
+        if not login:
+            raise InvalidAuth
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aladdin Connect."""
+
+    VERSION = 1
+    entry: config_entries.ConfigEntry | None
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication with Aladdin Connect."""
+
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm re-authentication with Aladdin Connect."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            try:
+                await validate_input(self.hass, user_input)
+            except ConnectionError:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            else:
+                assert self.entry is not None
+
+                if user_input[CONF_USERNAME].lower() == self.entry.unique_id:
+                    self.hass.config_entries.async_update_entry(
+                        self.entry,
+                        data={
+                            **self.entry.data,
+                            CONF_PASSWORD: user_input[CONF_PASSWORD],
+                        },
+                    )
+                    await self.hass.config_entries.async_reload(self.entry.entry_id)
+                    return self.async_abort(reason="reauth_successful")
+                errors["base"] = "invalid_auth"
+
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=STEP_USER_DATA_SCHEMA,
```

Comment:
```suggestion
            data_schema=vol.Schema(
                {
                    vol.Required(CONF_PASSWORD): str,
                }
            )
```
You don't need to show the user the username as they should not be allowed to change it anyhow as it's the unique id for the integration